flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/api/internal/TableEnvImpl.scala

```diff
@@ -73,7 +74,14 @@ abstract class TableEnvImpl(
     new TableReferenceLookup {
       override def lookupTable(name: String): Optional[TableReferenceExpression] = {
         JavaScalaConversionUtil
-          .toJava(scanInternal(Array(name)).map(t => new TableReferenceExpression(name, t)))
+          .toJava(
+            Try({
+              val unresolvedIdentifier = UnresolvedIdentifier.of(name)
```

Comment:
Why use `Try`? If it is an illegal name, should we throw exception?

Reply:
This gives positive results when resolving `IN` expression. But will be invoked for all `UnresolvedExpressions`. That's why it might not necessary be a table identifier. The contract of the `ExpressionResolver` is that it actually never throws exception.

As a side note. The way `TableIdentifier` is currently implemented is actually wrong. This method is primarily for resolving `SELECT * FROM t WHERE f IN (tableName)`, which is actually incorrect SQL statement. It is not allowed to use identifiers in an `IN` clause. Therefore once we introduce the Java Expression DSL we should get rid of this catalog lookup for a `TableReference`.

Reply:
Got it. You can put these comments to code.